lib/transaction/trace/index.js

```diff
@@ -218,13 +218,13 @@ Trace.prototype.getTotalTimeDurationInMillis = function getTotalTimeDurationInMi
   if (this.root.children.length === 0) {
     return 0
   }
-  let segments = this.root.getChildren()
+  const segments = this.root.getChildren()
   let totalTimeInMillis = 0
 
   while (segments.length !== 0) {
     const segment = segments.pop()
     totalTimeInMillis += segment.getExclusiveDurationInMillis()
-    segments = segments.concat(segment.getChildren())
+    segment.getChildren().forEach((childSegment) => segments.push(childSegment))
```

Comment:
can't this be improved further by using simple loop instead of a `forEach`? callback-based looping methods are noticeably slower.

Reply:
Probably. I've never studied the code. PRs are welcomed @kibertoad!

Reply:
@kibertoad you are correct ! Actually, at time of writing this code, I had verified: `forEach` had been already used at lot of places in agent code, and major issue was due to O(n^2) complexity order, that's why just solved that issue using forEach loop. `for` loop usage will not change the complexity order, but yeah, will improve it a bit more. Separate PR can be raised for same. Thanks